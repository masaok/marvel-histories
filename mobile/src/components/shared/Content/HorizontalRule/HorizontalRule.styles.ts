import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  horizontalRule: {
    borderBottomColor: 'lightgrey',
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
});
