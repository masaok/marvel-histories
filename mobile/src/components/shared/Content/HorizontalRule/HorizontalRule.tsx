/**
 * HorizontalRule
 */

import * as React from 'react';
import { Text, View, StyleProp, ViewStyle } from 'react-native';

import styles from './HorizontalRule.styles';

export interface Props {
  style?: StyleProp<ViewStyle>;
}

interface State {}

export default class HorizontalRule extends React.Component<Props, State> {
  render() {
    return <View style={[this.props.style, styles.horizontalRule]} />;
  }
}
