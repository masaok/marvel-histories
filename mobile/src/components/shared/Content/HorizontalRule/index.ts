import HorizontalRule from "./HorizontalRule";
export default HorizontalRule;
