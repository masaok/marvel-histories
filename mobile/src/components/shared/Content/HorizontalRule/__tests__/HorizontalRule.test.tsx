import * as React from "react";
import renderer from "react-test-renderer";

import HorizontalRule from "../HorizontalRule";

it("renders correctly", () => {
  const tree = renderer.create(<HorizontalRule />).toJSON();
  expect(tree).toMatchSnapshot();
});
