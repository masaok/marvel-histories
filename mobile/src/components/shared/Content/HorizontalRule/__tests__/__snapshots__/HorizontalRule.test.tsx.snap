// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "borderBottomColor": "lightgrey",
        "borderBottomWidth": 0.5,
      },
    ]
  }
/>
`;
